feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go

```diff
@@ -0,0 +1,641 @@
+				Encoding:    certzpb.CertificateEncoding_CERTIFICATE_ENCODING_PEM,
+				Certificate: serverCertContent,
+				PrivateKey:  serverKeyContent}, Parent: nil}
+		}
+		return certzpb.CertificateChain{Certificate: &certzpb.Certificate{
+			Type:        certzpb.CertificateType_CERTIFICATE_TYPE_X509,
+			Encoding:    certzpb.CertificateEncoding_CERTIFICATE_ENCODING_PEM,
+			Certificate: serverCertContent,
+			PrivateKey:  nil}, Parent: nil}
+
+	case EntityTypeTrustBundle:
+		if len(certData.TrustBundleFile) == 0 {
+			t.Fatalf("Missing trust bundle file for creating certificate chain object.")
+		}
+		trustBundleContent, err := os.ReadFile(certData.TrustBundleFile)
+		if err != nil {
+			t.Fatalf("Error reading trust bundle file at: %v with error: %v", certData.TrustBundleFile, err)
+		}
+		return certzpb.CertificateChain{Certificate: &certzpb.Certificate{
+			Type:        certzpb.CertificateType_CERTIFICATE_TYPE_X509,
+			Encoding:    certzpb.CertificateEncoding_CERTIFICATE_ENCODING_PEM,
+			Certificate: trustBundleContent,
+		}, Parent: nil}
+
+	default:
+		t.Fatalf("Invalid request type received.")
+	}
+	return certzpb.CertificateChain{}
+}
+
+// CreateCertChainFromTrustBundle function to create the certificate chain from trust bundle.
+func CreateCertChainFromTrustBundle(fileName string) *certzpb.CertificateChain {
+	pemData, err := os.ReadFile(fileName)
+	if err != nil {
+		return &certzpb.CertificateChain{}
+	}
+	var trust [][]byte
+	for {
+		var block *pem.Block
+		block, pemData = pem.Decode(pemData)
+		if block == nil {
+			break
+		}
+		if block.Type != "CERTIFICATE" {
+			continue
+		}
+		p := pem.EncodeToMemory(block)
+		if p == nil {
+			return &certzpb.CertificateChain{}
+		}
+		trust = append(trust, p)
+	}
+	//a valid check for trust not empty
+	if len(trust) == 0 {
+		return &certzpb.CertificateChain{}
+	} else {
+		var prevCert *certzpb.CertificateChain
+		var bundleToReturn *certzpb.CertificateChain
+		for i := len(trust) - 1; i >= 0; i-- {
+			if i == len(trust)-1 {
+				bundleToReturn = &certzpb.CertificateChain{Certificate: &certzpb.Certificate{
+					Type:        certzpb.CertificateType_CERTIFICATE_TYPE_X509,
+					Encoding:    certzpb.CertificateEncoding_CERTIFICATE_ENCODING_PEM,
+					Certificate: trust[i],
+				}, Parent: nil}
+				prevCert = bundleToReturn
+			} else {
+				prevCert = bundleToReturn
+				bundleToReturn = &certzpb.CertificateChain{Certificate: &certzpb.Certificate{
+					Type:        certzpb.CertificateType_CERTIFICATE_TYPE_X509,
+					Encoding:    certzpb.CertificateEncoding_CERTIFICATE_ENCODING_PEM,
+					Certificate: trust[i],
+				}, Parent: prevCert}
+			}
+		}
+		return bundleToReturn
+	}
+}
+
+// CertzRotate function to request the client certificate rotation and returns true on successful rotation.
+func CertzRotate(t *testing.T, caCert *x509.CertPool, certzClient certzpb.CertzClient, cert tls.Certificate, san, serverAddr, profileID string, entities ...*certzpb.Entity) bool {
```

Comment:
There are unused arguments for this function -- since this isn't meeting a signature defined elsewhere, please remove the arguments.